core/block/backlinks/watcher.go

```diff
@@ -88,9 +88,9 @@ func (w *watcher) Close(context.Context) error {
 	return nil
 }
 
-func (w *watcher) Run(context.Context) error {
+func (w *watcher) Run(ctx context.Context) error {
 	w.updater.SubscribeLinksUpdate(func(info spaceindex.LinksUpdateInfo) {
-		if err := w.infoBatch.Add(info); err != nil {
+		if err := w.infoBatch.Add(ctx, info); err != nil {
```

Comment:
you should use another context, ideally -- create component context and cancel it in Close method